code/modules/luck/luck.dm

```diff
@@ -0,0 +1,156 @@
+	//blesscurse: List of blessings and curses: see luck_blesscurse.dm.
+/datum/luckiness
+	var/permanent_luckiness = 0
+	var/temporary_luckiness = 0
+	var/list/blesscurse = list()
+
+//Returns a mob's base luckiness.
+/datum/luckiness/proc/base_luck()
+	var/base_luck = permanent_luckiness + temporary_luckiness
+	if(blesscurse.len)
+		for(var/datum/blesscurse/this_blesscurse in blesscurse)
+			base_luck += this_blesscurse.blesscurse_strength
+	return base_luck
+
+//Add a blessing or curse to a mob.
+/mob/proc/add_blesscurse(var/datum/blesscurse/ourblesscurse)
+	if(!base_luck)
+		base_luck = new /datum/luckiness
+	if(has_blesscurse(ourblesscurse)) //can only have one instance of each type of blessing or curse
+		return
+	base_luck.blesscurse += ourblesscurse
+
+//Remove a blessing or curse from a mob.
+/mob/proc/remove_blesscurse(var/datum/blesscurse/ourblesscurse)
+	if(!base_luck)
+		return
+	base_luck.blesscurse -= ourblesscurse
+
+//Check if a mob has a given type of blessing or curse.
+/mob/proc/has_blesscurse(var/datum/blesscurse/ourblesscurse)
+	if(!base_luck)
+		return FALSE
+	if(base_luck.blesscurse.len)
+		for(var/i in 1 to base_luck.blesscurse.len)
+			if(ispath(base_luck.blesscurse[i], ourblesscurse))
+				return TRUE
+	else
+		return FALSE
+
+//Alters a mob's base luckiness.
+/mob/proc/luck_adjust(var/luckchange, var/temporary = FALSE)
+	if(!base_luck)
+		base_luck = new /datum/luckiness
+	if(temporary)
+		base_luck.temporary_luckiness += luckchange
+	else
+		base_luck.permanent_luckiness += luckchange
+
+//Returns the total luckiness of all items held or worn by the mob. Items confer luckiness in different inventory locations according to their luckiness_validity flags: see setup.dm.
+/mob/proc/borne_item_luckiness() //Check all inventory items for (un)luckiness.
+	var/total_item_luckiness = 0
+	var/list/equipped_items = get_equipped_items()
+	while(null in equipped_items)
+		equipped_items -= null
+	var/list/held_or_implanted_items = contents - equipped_items
+	//check hand slot items and implanted items
+	if(held_or_implanted_items.len)
+		for(var/obj/item/thisitem in held_or_implanted_items)
+			if(thisitem.luckiness_validity & (LUCKINESS_WHEN_HELD | LUCKINESS_WHEN_HELD_RECURSIVE) && thisitem.luckiness)
+				total_item_luckiness += thisitem.luckiness
+			//check hand slot and implanted item contents recursively
+			if(thisitem.contents.len)
+				var/list/thisitemscontents = thisitem.contents
+				var/j = 1
+				while(j <= thisitemscontents.len)
+					var/obj/item/thiscontent = thisitemscontents[j]
+					if(thiscontent.contents.len)
+						thisitemscontents += thiscontent.contents
+					j += 1
+				for(var/obj/item/thisitemscontent in thisitemscontents)
+					if(thisitemscontent.luckiness_validity & LUCKINESS_WHEN_HELD_RECURSIVE && thisitemscontent.luckiness)
+						total_item_luckiness += thisitemscontent.luckiness
+	//check worn items directly
+	if(equipped_items.len)
+		for(var/obj/item/thisitem in  equipped_items.len)
```

Comment:
Think you made a mistake in changing this one. All the others look good at a glance. This really shouldn't compile but lol byond